src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheckTest.java

```diff
@@ -74,4 +74,24 @@ public void testTokensNotNull() {
         Assert.assertNotNull(check.getDefaultTokens());
         Assert.assertNotNull(check.getRequiredTokens());
     }
+
+    @Test
+    public void testonlyObjectReferences() throws Exception {
+        final DefaultConfiguration checkConfig =
+            createCheckConfig(ExplicitInitializationCheck.class);
+        checkConfig.addAttribute("onlyObjectReferences", "true");
+        final String[] expected = {
+            "5:20: " + getCheckMessage(MSG_KEY, "bar", "null"),
+            "14:22: " + getCheckMessage(MSG_KEY, "str1", "null"),
+            "14:35: " + getCheckMessage(MSG_KEY, "str3", "null"),
+            "15:9: " + getCheckMessage(MSG_KEY, "ar1", "null"),
```

Comment:
@romani We consider all arrays, including primitives, to be objects?
This seems slightly weird, but makes sense since it can be assigned `null`. I'm not sure if issue creator realized this.

I think we should mention this in the xdoc if we keep this as is.